analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSyntaxHelper.cs

```diff
@@ -282,8 +282,23 @@ node switch
         public static bool HasConstantValue(this ExpressionSyntax expression, SemanticModel semanticModel) =>
             expression.RemoveParentheses().IsAnyKind(LiteralSyntaxKinds) || expression.FindConstantValue(semanticModel) != null;
 
-        public static string GetStringValue(this SyntaxNode node, SemanticModel semanticModel) =>
-            CsharpStringValueHelper.Instance.GetStringValue(node, semanticModel);
+        public static string StringValue(this SyntaxNode node, SemanticModel semanticModel)
+        {
+            if (node != null)
+            {
+                if (node.IsAnyKind(SyntaxKind.StringLiteralExpression, SyntaxKindEx.Utf8StringLiteralExpression)
+                    && node is LiteralExpressionSyntax literal)
```

Comment:
What is the `IsAnyKind` check for, when we have `node is LiteralExpressionSyntax`?

Reply:
We want only to check for literals but only of string type.
If we don't do this check also for example `x = null` is being picked up.